post_info.php

```diff
@@ -0,0 +1,695 @@
+<?php
+require 'post_query_generators.php';
+require 'db/config.php';
+// header("Content-type: application/json");
+// below is the block for receiving POST request from the frontend
+try {
+    $GLOBALS["conn"] = new PDO(
+        "mysql:host=$servername;dbname=$database",
+        $username,
+        $password
+    );
+    // Get the operations that the user wants to perform
+    $action = $_REQUEST['action'];
+    // Check the validation of $action
+    if (
+        $action != "add_or_update" &&
+        $action != "branch" &&
+        $action != "delete"
+    ) {
+        throw new Exception("'action' attribute not valid!");
```

Comment:
Make this more specific. Like `"'action' attribute '$action' unknown"`

Reply:
Sure!